hummingbot/strategy/pure_market_making/pure_market_making.pyx

```diff
@@ -974,27 +993,27 @@ cdef class PureMarketMakingStrategy(StrategyBase):
                 return
 
         cdef:
-            object mid_price = self.c_get_mid_price()
+            object price = self.get_price()
             list active_orders = self.active_orders
             list orders
             LimitOrder order
         for h_order_id in self._hanging_order_ids:
             orders = [o for o in active_orders if o.client_order_id == h_order_id]
-            if orders and mid_price > 0:
+            if orders and price > 0:
                 order = orders[0]
-                if abs(order.price - mid_price)/mid_price >= self._hanging_orders_cancel_pct:
+                if abs(order.price - price)/price >= self._hanging_orders_cancel_pct:
                     self.c_cancel_order(self._market_info, order.client_order_id)
 
     # Cancel Non-Hanging, Active Orders if Spreads are below minimum_spread
     cdef c_cancel_orders_below_min_spread(self):
         cdef:
             list active_orders = self.market_info_to_active_orders.get(self._market_info, [])
-            object mid_price = self._market_info.get_mid_price()
+            object price = self.get_price()  # TODO: check if this was previously a bug or intended
```

Comment:
Not sure if this was purposefully set to ignore the external market, if one is set.

Reply:
You're right, this is a bug. It should be `get_price`